File: solutions/217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python
#
# [217] Contains Duplicate
#

# @lc code=start
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        store = set()
        for i in nums:
            if i in store:
                return True
            else:
                store.add(i)
                return False
        
# @lc code=end

File: solutions/217.contains-duplicate.java
/*
 * @lc app=leetcode id=217 lang=java
 *
 * [217] Contains Duplicate
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> counter = new HashSet<>();
        for (int i : nums) {
            if (counter.contains(i)) {
                return true;
            } else {
                counter.add(i);
            }
        }
        return false;
    }
}
// @lc code=end

